File: backend/smart-library-backend/src/main/java/com/library/service/UserService.java
package com.library.service;

import com.library.dto.request.ChangePasswordRequest;
import com.library.dto.request.UpdateProfileRequest;
import com.library.dto.response.PageResponse;
import com.library.dto.response.UserResponse;
import org.springframework.data.domain.Pageable;

public interface UserService {
    PageResponse<UserResponse> getAllStudents(String keyword, Pageable pageable);
    UserResponse getStudentById(Long id);
    UserResponse updateStudent(Long id, UpdateProfileRequest request);
    void deactivateStudent(Long id);
    void activateStudent(Long id);
    UserResponse updateMyProfile(String username, UpdateProfileRequest request);
    void changePassword(String username, ChangePasswordRequest request);
}

File: frontend/smart-library-frontend/src/services/userService.js

import api from './api';

export const getAllStudents = (page = 0, size = 10, keyword = '') =>
  api.get(`/users/students?page=${page}&size=${size}${keyword ? '&keyword=' + encodeURIComponent(keyword) : ''}`);
export const getStudentById = (id) => api.get(`/users/students/${id}`);
export const updateStudent = (id, data) => api.put(`/users/students/${id}`, data);
export const deactivateStudent = (id) => api.put(`/users/students/${id}/deactivate`);
export const activateStudent = (id) => api.put(`/users/students/${id}/activate`);
export const getMyProfile = () => api.get('/users/profile');
export const updateMyProfile = (data) => api.put('/users/profile', data);
export const changePassword = (data) => api.put('/users/change-password', data);
